lib/web/sessions.go

```diff
@@ -850,6 +850,19 @@ func (s *sessionCache) invalidateSession(ctx context.Context, sctx *SessionConte
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	if err := clt.DeleteUserAppSessions(ctx, &proto.DeleteUserAppSessionsRequest{Username: sctx.GetUser()}); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if sctx.cfg.Session.GetSAMLSession() != nil {
+		if err := clt.DeleteSAMLIdPSession(ctx, types.DeleteSAMLIdPSessionRequest{
+			SessionID: sctx.cfg.Session.GetSAMLSession().ID,
+		}); err != nil && !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+	}
```

Comment:
Suggestion: Use the if scope to cut repetition, check the ID as well.

```suggestion
	if session := sctx.cfg.Session.GetSAMLSession(); session != nil && session.ID != "" {
		if err := clt.DeleteSAMLIdPSession(ctx, types.DeleteSAMLIdPSessionRequest{
			SessionID: session.ID,
		}); err != nil && !trace.IsNotFound(err) {
			return trace.Wrap(err)
		}
	}
```